File: Swift-C++-Testsuite/test/extensions/Swift/NestedEnum.swift
@main
struct MyMain {

    static func main() {
        func printIntegerKinds(_ numbers: [Int]) {
            for number in numbers {
                switch number.kind {
                case .negative:
                    print("- ", terminator: "")
                case .zero:
                    print("0 ", terminator: "")
                case .positive:
                    print("+ ", terminator: "")
                }
            }
            print("")
        }
        printIntegerKinds([3, 19, -27, 0, -6, 0, 7])
    }
}

extension Int {
    enum Kind {
        case negative, zero, positive
    }
    var kind: Kind {
        switch self {
        case 0:
            return .zero
        case let x where x > 0:
            return .positive
        default:
            return .negative
        }
    }
}

File: AutoTranspiledTests/try-1/NestedEnum.cpp
 
#include <iostream>
#include <vector>

class MyMain {
    public:
        static void main()
        {
            void printIntegerKinds(std::vector<int> numbers) 
            { 
                for(int number : numbers) 
                { 
                    switch (number.kind())
                    {
                        case negative: 
                            std::cout << "-";
                            break;
                        case zero:
                            std::cout <<"0"; 
                            break;
                        case positive:
                            std::cout << "+";
                            break;
                    } 
                } 
                std::cout << std::endl;
            } 
            
            printIntegerKinds({3, 19, -27, 0, -6, 0, 7});
        }

    private: 
        enum Kind {negative, zero, positive};
    
        Kind kind() {
            switch (this)
            {
                case 0: 
                    return zero;
                case (let x) > 0: 
                    return positive;
                default:
                    return negative;
            }
        } 
}; 

int main() { 
    MyMain::main(); 
    return 0; 
}